src/lib/InstantSearch.ts

```diff
@@ -339,7 +339,7 @@ See ${createDocumentationLink({
    * Widgets can be added either before or after InstantSearch has started.
    * @param widgets The array of widgets to add to InstantSearch.
    */
-  public addWidgets(widgets: Widget[]) {
+  public addWidgets(widgets: Array<Widget | Index>) {
```

Comment:
This gives me the impression that `index` is not a widget at all, whereas in fact it's a sort of special widget.

```suggestion
  public addWidgets(widgets: Array<Widget | IndexWidget>) {
```

I think it aligns better with other widgets like `HitsWidget, RefinementListWidget, ...` and also with `isIndexWidget()` function.

Reply:
makes sense, let's do this while it's not yet a breaking change